### PR TITLE
Added support for authentication codes for the VM service

### PR DESCRIPTION
**Pull request by bkonyi**
Previously, a valid web socket connection would use the following URI:

`ws://127.0.0.1/ws`

Now, if enabled, the VM service requires a connection to be made with a
URI similar to the following:

`ws://127.0.0.1:8181/Ug_U0QVsqFs=/ws`

where `Ug_U0QVsqFs` is an authentication code generated and shared by the
service.

This behavior can be enabled with the `--enable-service-auth-codes`
flag.

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/FlutterShellArgs.java [hunk 1]
```diff
@@ -133,4 +138,4 @@ public void remove(@NonNull String arg) {
     String[] argsArray = new String[args.size()];
     return args.toArray(argsArray);
   }
-}
\ No newline at end of file
+}
```
Comment (bkonyi) [inline shell/platform/android/io/flutter/embedding/engine/FlutterShellArgs.java]: Not sure why this line changed, but I'm guessing it's something to do with Vim. I tried to fix it but wasn't having much luck. If it's a problem, I'll go and fight with it some more.

Comment (zanderso) [reply shell/platform/android/io/flutter/embedding/engine/FlutterShellArgs.java]: Your editor (or maybe git) is adding a newline at the end of the file.

File: common/settings.h [hunk 2]
```diff
@@ -86,6 +86,10 @@ struct Settings {
   uint32_t observatory_port = 0;
   bool ipv6 = false;
 
+  // Determines whether an authentication code is required to communicate with
+  // the VM service.
+  bool disable_service_auth_codes = true;
```
Comment (chinmaygarde) [inline common/settings.h]: Can this not be a negative please? `enable_service_auth_codes = false`

Comment (bkonyi) [reply common/settings.h]: This is meant to be temporary and will be enabled by default later. I'd really rather avoid having to change everything only to have to change it back in the next day or so.

**Comment by bkonyi** [comment]
Thanks!

**Pull request merged**